pkg/reconciler/core/logicalclusterdeletion/logicalcluster_deletion_controller.go

```diff
@@ -236,49 +245,15 @@ func (c *Controller) process(ctx context.Context, key string) error {
 		return c.finalizeWorkspace(ctx, logicalClusterCopy)
 	}
 
-	if err := c.patchCondition(ctx, logicalCluster, logicalClusterCopy); err != nil {
-		return err
-	}
-
-	return deleteErr
-}
-
-func (c *Controller) patchCondition(ctx context.Context, old, new *corev1alpha1.LogicalCluster) error {
-	logger := klog.FromContext(ctx)
-	if equality.Semantic.DeepEqual(old.Status.Conditions, new.Status.Conditions) {
-		return nil
-	}
-
-	oldData, err := json.Marshal(corev1alpha1.LogicalCluster{
-		Status: corev1alpha1.LogicalClusterStatus{
-			Conditions: old.Status.Conditions,
-		},
-	})
-	if err != nil {
-		return fmt.Errorf("failed to Marshal old data for logical cluster %s: %w", old.Name, err)
-	}
-
-	newData, err := json.Marshal(corev1alpha1.LogicalCluster{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:             old.UID,
-			ResourceVersion: old.ResourceVersion,
-		}, // to ensure they appear in the patch as preconditions
-		Status: corev1alpha1.LogicalClusterStatus{
-			Conditions: new.Status.Conditions,
-		},
-	})
-	if err != nil {
-		return fmt.Errorf("failed to Marshal new data for logical cluster %s: %w", new.Name, err)
-	}
+	errs := []error{deleteErr}
 
-	patchBytes, err := jsonpatch.CreateMergePatch(oldData, newData)
-	if err != nil {
-		return fmt.Errorf("failed to create patch for logical cluster %s: %w", new.Name, err)
+	oldResource := &Resource{ObjectMeta: logicalCluster.ObjectMeta, Spec: &logicalCluster.Spec, Status: &logicalCluster.Status}
+	newResource := &Resource{ObjectMeta: logicalClusterCopy.ObjectMeta, Spec: &logicalClusterCopy.Spec, Status: &logicalClusterCopy.Status}
+	if err := c.commit(ctx, oldResource, newResource); err != nil {
+		errs = append(errs, err)
 	}
 
-	logger.V(2).Info("patching LogicalCluster", "patch", string(patchBytes))
-	_, err = c.kcpClusterClient.Cluster(logicalcluster.From(new).Path()).CoreV1alpha1().LogicalClusters().Patch(ctx, new.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
```

Comment:
@ncdc the logicalcluster used to scope the patch here is gotten from the `new` resource whereas in the committer it uses the old resource: https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/committer/committer.go#L63-L64. Should committer use the new resource instead?